caddyhttp/proxy/reverseproxy.go

```diff
@@ -89,6 +88,18 @@ func socketDial(hostName string) func(network, addr string) (conn net.Conn, err
 	}
 }
 
+func singleJoiningSlash(a, b string) string {
+	aslash := strings.HasSuffix(a, "/")
+	bslash := strings.HasPrefix(b, "/")
+	switch {
+	case aslash && bslash:
+		return a + b[1:]
+	case !aslash && !bslash && b != "":
+		return a + "/" + b
+	}
+	return a + b
+}
```

Comment:
`path.Join` will also strip the middle slash: `/abc/` + `/de` = `abc/de` instead of `abc//de`. This case will often occur in the encoded URL and it is the way that [standard lib uses](https://github.com/golang/go/blob/master/src/net/http/httputil/reverseproxy.go#L95).